src/moscot/problems/base/_mixins.py

```diff
@@ -476,3 +478,100 @@ def _cell_aggregation_transition(
+
+        Correlates (processed) count data with probabilities of cells mapped to a subset of cells, e.g. for a
+        certain celltype.
+
+        Parameters
+        ----------
+        obs_key
+            Column of :attr:`adata.obs` containing push-forward or pull-back distributions.
+        method
+            Mode to use when calculating p-values and confidence intervals. Valid options are:
+                - `{CorrTestMethod.FISCHER!r}` - use Fischer transformation :cite:`fischer:21`.
+                - `{CorrTestMethod.PERM_TEST!r}` - use permutation test.
+        annotation
+            If not `None`, this defines the subset of data to be considered when computing the correlation.
+            Its key should correspond to a key in
+            :attr:`anndata.AnnData.obs` and its value to a list containing a subset of categories present in
+            :attr:`anndata.AnnData.obs` ``['{annotation.keys()[0]}']``.
+        layer
+            Key from :attr:`anndata.AnnData.layers` from which to get the expression.
+            If `None`, use :attr:`anndata.AnnData.X`.
+        features
+            Features of :class:`anndata.AnnData` which the correlation of ``anndata.AnnData.obs['{obs_key}']`` is
+            computed with:
+            - If `None`, all features will be taken into account.
+            - If of type :obj:`list`, features from :attr:`anndata.AnnData.var_names` will be taken.
+            - If `human`, `mouse`, or `drosophila`, the features are subsetted to transcription factors.
+        confidence_level
+            Confidence level for the confidence interval calculation. Must be in interval `[0, 1]`.
+        n_perms
+            Number of permutations to use when ``method = {CorrTestMethod.PERM_TEST!r}``.
+        seed
+            Random seed when ``method = {CorrTestMethod.PERM_TEST!r}``.
+
+        %(parallel_kwargs)s
+
+        Returns
+        -------
+        %(correlation_test.returns)s
+        """
+        if obs_key not in self.adata.obs.columns:
+            raise KeyError(f"[{obs_key!r} not found in `adata.obs`.")
+
+        method = CorrTestMethod(method)
+
+        if annotation is not None:
+            annotation_key, annotation_vals = next(iter(annotation.items()))
+            if annotation_key not in self.adata.obs:
+                raise KeyError(f"[{annotation_key!r} not found in `adata.obs`.")
+            if not isinstance(annotation_vals, list):
+                raise TypeError("`annotation` expected to be dictionary of length 1 with value being a list.")
+
+            adata_red = self.adata[self.adata.obs[annotation_key].isin(annotation_vals)]
+        else:
+            adata_red = self.adata
+
+        adata_red = adata_red[~adata_red.obs[obs_key].isnull()]
+        if len(adata_red) == 0:
+            raise ValueError(f"`adata.obs[{obs_key!r}]` only contains NaN values.")
+        distribution = adata_red.obs[[obs_key]]
```

Comment:
Add a comment please that you expect `distribution` to be a data frame instead of a series.

Reply:
Come to think of this, why it's `(n_cells, 1)` only? There can be many things correlated in the original implementation (this slightly limits the use of `obs_key`, which must be a string, but it could be a sequence of strings in `adata.obs`).

Reply:
In `CellRank`, I only return 1 dataframe that has many columns (`k x 5`, where `k` is the number of columns in `y` and `5` is because of the corr, p-val, q-val, ci-low, ci-high).

Reply:
Then it becomes more complicated with sorting, because then by default it's sorted w.r.t. a certain correlation value which might be misleading IMO. Don't you think so?

Open to have `obs_key` as a list but might be slightly too complicated I think. If you have a strong opinion on allowing it, I can implement it.